Support phps' docref_root ini-param. Fixes #499.

staabm opened the pull request:
when php-ini `html_errors` and `docref_root` is defined we parse the url to the manual page out of the error message and link it with a book icon (besides google and stackoverflow icon)

![image](https://cloud.githubusercontent.com/assets/120441/26732121/c6165fa0-47b6-11e7-9418-ad0aacea9d20.png)

staabm commented:
just a  prototype how #499 coud be supported.

IMO we should also parse the actual exception message and strip all the part in square brackets, so the exception message looks like when those 2 php ini settings are not defined... wdyt?

staabm commented on src/Whoops/Exception/Inspector.php:
```diff
@@ -92,45 +112,45 @@ public function getPreviousExceptionInspector()
-            $frames = $this->getTrace($this->exception);            
-            
-            // Fill empty line/file info for call_user_func_array usages (PHP Bug #44428) 
+            $frames = $this->getTrace($this->exception);
```
arghs when did we introduce this whitespace crap?

staabm replied:
fixed in https://github.com/filp/whoops/pull/502/files .. will rebase after #502 landed

denis-sokolov replied:
Landed.

staabm commented:
//cc @jens1o

jens1o commented:
Looks good, but could we remove the link? This doesn't look that nice to be honest.

staabm commented:
yeah, thats also my opinion.

> IMO we should also parse the actual exception message and strip all the part in square brackets, so the exception message looks like when those 2 php ini settings are not defined... wdyt?

lets the what the maintainer of this project thinks... will do whatever he likes to get this landed.

jens1o commented:
Yeah, I'm sorry, I didn't read that.

denis-sokolov commented:
@staabm, you have my full support!

jens1o commented on src/Whoops/Resources/views/header.html.php:
```diff
@@ -20,6 +20,16 @@
+          <svg height="16px" id="Layer_1" style="enable-background:new 0 0 32 32;" version="1.1" viewBox="0 0 32 32" width="16px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><g transform="translate(240 0)"><path d="M-211,4v26h-24c-1.104,0-2-0.895-2-2s0.896-2,2-2h22V0h-22c-2.209,0-4,1.791-4,4v24c0,2.209,1.791,4,4,4h26V4H-211z    M-235,8V2h20v2 ...
+        </a>
+      </li>
+      <?php endif ?>
```
You are missing a semicolon? `<?php endif; ?>`

staabm replied:
we do this consistently in this file that way.

staabm commented:
fixed and rebased... the links are stripped from the exception message but linked from the icon below as suggested earlier:

![image](https://cloud.githubusercontent.com/assets/120441/26753004/1fffbe70-485d-11e7-9766-a2a406f2f6bb.png)


ready to land

jens1o commented:
lgtm

Can't wait for a release.

staabm commented:
valid points... adjusted per feedback.

denis-sokolov commented:
Thank you, @staabm!

I understand we will also parse URLs that are not to php.net, but I think it’s still a good idea. Releasing a new version...

denis-sokolov commented:
2.1.9!

staabm commented:
> I understand we will also parse URLs that are not to php.net, but I think it’s still a good idea. Releasing a new version...

The native php feature works also with downloaded and selfhosted versions of the manual... and this is also what this PR supports.. so I think it works like expected

jens1o commented:
Thanks to all of you. c:

the94air commented:
Guys, get ready for Laravel 5.5. really nice work :+1: 
https://laravel-news.com/whoops-laravel-5-5
